content/src/service/synchronization/SynchronizationManager.ts

```diff
@@ -23,15 +24,40 @@ export class ClusterSynchronizationManager implements SynchronizationManager {
   private synchronizationState: SynchronizationState = SynchronizationState.BOOTSTRAPPING
   private stopping: boolean = false
   private timeOfLastSync: Timestamp = 0
+  private timeoutStreamId: NodeJS.Timeout
 
   constructor(
     private readonly cluster: ContentCluster,
     private readonly systemProperties: SystemPropertiesManager,
     private readonly deployer: EventDeployer,
     private readonly timeBetweenSyncs: number,
-    private readonly disableSynchronization: boolean
+    private readonly disableSynchronization: boolean,
+    private readonly syncStreamTimeout: string
   ) {}
 
+  timeoutStreams(streams: Transform[]): void {
+    this.timeoutStreamId = setTimeout(() => {
```

Comment:
Since this function is called from an async function, I'd suggest to return the timer instead of using the `this` object to prevent the logic from breaking due to asynchronous conditions.

Also, I'd create a free function, avoiding methods when possible